[OpenAPI] Edits connector API summaries

lcawl opened the pull request:
Relates to https://github.com/elastic/elasticsearch-specification/issues/2635

This PR updates the operation summaries and a few operation descriptions with content copied from https://www.elastic.co/guide/en/elasticsearch/reference/master/connector-apis.html

NOTE: I didn't see a difference in the old docs between the "update filtering" and "update active filtering", so I left the text as-is in the latter.

leemthompo commented on docs/overlays/elasticsearch-shared-overlays.yaml:
```diff
@@ -36,6 +36,7 @@ actions:
             Connectors are Elasticsearch integrations that bring content from third-party data sources, which can be deployed on Elastic Cloud or hosted on your own infrastructure:
             
             * Native connectors are a managed service on Elastic Cloud
+            
             * Connector clients are self-managed on your infrastructure
```
```suggestion
            * Elastic managed connectors (Native connectors) are a managed service on Elastic Cloud
            
            * Self-managed connectors (Connector clients) are self-managed on your infrastructure
```

leemthompo replied:
🚗We're trying to simplify the taxonomy and phase out these slightly confusing original names

leemthompo commented on output/schema/schema.json:
```diff
@@ -3074,7 +3074,7 @@
         }
       },
-      "description": "Creates a connector.",
+      "description": "Create a connector.\n\nConnectors are Elasticsearch integrations that bring content from third-party data sources, which can be deployed on Elastic Cloud or hosted on your own infrastructure.\nNative connectors are a managed service on Elastic Cloud.\nConnector clients are self-managed on your infrastructure.",
```
```suggestion
      "description": "Create a connector.\n\nConnectors are Elasticsearch integrations that bring content from third-party data sources, which can be deployed on Elastic Cloud or hosted on your own infrastructure.\nElastic managed connectors (Native connectors) are a managed service on Elastic Cloud.\nSelf-managed connectors (Connector clients) are self-managed on your infrastructure.",
```

lcawl replied:
For future reference, reviewing the output folder is not necessary, since those changes are generated from the other files. When we have automated previews it'll be even more unnecessary, since you will be able to see how it looks in the preview.

leemthompo commented on output/openapi/elasticsearch-openapi.json:
```diff
@@ -4382,7 +4387,8 @@
         ],
-        "summary": "Creates a connector",
+        "summary": "Create a connector",
+        "description": "Connectors are Elasticsearch integrations that bring content from third-party data sources, which can be deployed on Elastic Cloud or hosted on your own infrastructure.\nNative connectors are a managed service on Elastic Cloud.\nConnector clients are self-managed on your infrastructure.",
```
```suggestion
        "description": "Connectors are Elasticsearch integrations that bring content from third-party data sources, which can be deployed on Elastic Cloud or hosted on your own infrastructure.\nElastic managed connectors (Native connectors) are a managed service on Elastic Cloud.\nSelf-managed connectors (Connector clients) are self-managed on your infrastructure.",
```

leemthompo commented on output/openapi/elasticsearch-serverless-openapi.json:
```diff
@@ -1962,7 +1966,8 @@
         ],
-        "summary": "Creates a connector",
+        "summary": "Create a connector",
+        "description": "Connectors are Elasticsearch integrations that bring content from third-party data sources, which can be deployed on Elastic Cloud or hosted on your own infrastructure.\nNative connectors are a managed service on Elastic Cloud.\nConnector clients are self-managed on your infrastructure.",
```
```suggestion
        "description": "Connectors are Elasticsearch integrations that bring content from third-party data sources, which can be deployed on Elastic Cloud or hosted on your own infrastructure.\nElastic managed connectors (Native connectors) are a managed service on Elastic Cloud.\nSelf-managed connectors (Connector clients) are self-managed on your infrastructure.",
```

leemthompo commented on output/openapi/elasticsearch-serverless-openapi.json:
```diff
@@ -2330,7 +2340,8 @@
         ],
-        "summary": "Updates the API key id in the connector document",
+        "summary": "Update the connector API key ID",
+        "description": "Update the `api_key_id` and `api_key_secret_id` fields of a connector.\nYou can specify the ID of the API key used for authorization and the ID of the connector secret where the API key is stored.\nThe connector secret ID is required only for native connectors.\nConnector clients do not use this field.",
```
```suggestion
        "description": "Update the `api_key_id` and `api_key_secret_id` fields of a connector.\nYou can specify the ID of the API key used for authorization and the ID of the connector secret where the API key is stored.\nThe connector secret ID is required only for Elastic managed connectors.\nSelf-managed connectors do not use this field.",
```

leemthompo commented on output/openapi/elasticsearch-openapi.json:
```diff
@@ -4750,7 +4761,8 @@
         ],
-        "summary": "Updates the API key id in the connector document",
+        "summary": "Update the connector API key ID",
+        "description": "Update the `api_key_id` and `api_key_secret_id` fields of a connector.\nYou can specify the ID of the API key used for authorization and the ID of the connector secret where the API key is stored.\nThe connector secret ID is required only for native connectors.\nConnector clients do not use this field.",
```
```suggestion
        "description": "Update the `api_key_id` and `api_key_secret_id` fields of a connector.\nYou can specify the ID of the API key used for authorization and the ID of the connector secret where the API key is stored.\nThe connector secret ID is required only for Elastic managed connectors.\nSelf-managed connectors do not use this field.",
```

leemthompo commented on output/schema/schema.json:
```diff
@@ -3589,7 +3589,7 @@
         }
       },
-      "description": "Updates the API key id in the connector document",
+      "description": "Update the connector API key ID.\n\nUpdate the `api_key_id` and `api_key_secret_id` fields of a connector.\nYou can specify the ID of the API key used for authorization and the ID of the connector secret where the API key is stored.\nThe connector secret ID is required only for native connectors.\nConnector clients do not use this field.",
```
```suggestion
      "description": "Update the connector API key ID.\n\nUpdate the `api_key_id` and `api_key_secret_id` fields of a connector.\nYou can specify the ID of the API key used for authorization and the ID of the connector secret where the API key is stored.\nThe connector secret ID is required only for Elastic managed connectors.\nSelf-managed connectors do not use this field.",
```

leemthompo commented on output/schema/schema.json:
```diff
@@ -113891,7 +113891,7 @@
         ]
       },
-      "description": "Creates a connector.",
+      "description": "Create a connector.\n\nConnectors are Elasticsearch integrations that bring content from third-party data sources, which can be deployed on Elastic Cloud or hosted on your own infrastructure.\nNative connectors are a managed service on Elastic Cloud.\nConnector clients are self-managed on your infrastructure.",
```
```suggestion
      "description": "Create a connector.\n\nConnectors are Elasticsearch integrations that bring content from third-party data sources, which can be deployed on Elastic Cloud or hosted on your own infrastructure.\nElastic managed connectors (Native connectors) are a managed service on Elastic Cloud.\nSelf-managed connectors (Connector clients) are self-managed on your infrastructure.",
```

leemthompo reviewed (approved):
LGTM with some drive-by taxonomy updates!

One question: the connectors docs have been moved to the [Elasticsearch guide in 8.16](https://www.elastic.co/guide/en/elasticsearch/reference/8.16/es-connectors.html). Because links are full URLs and they use `current` I'm guessing we can't make those changes until the version bump happen?

So we'll need to update [this line](https://github.com/elastic/elasticsearch-specification/pull/3051/files#diff-05d0423d9f49f7cc5ca19cf4319d27dd211b9b48349e6de5548765bb212bdd19R44) (and probably more links in the generated ref) when the time comes.

lcawl commented:
> One question: the connectors docs have been moved to the [Elasticsearch guide in 8.16](https://www.elastic.co/guide/en/elasticsearch/reference/8.16/es-connectors.html). Because links are full URLs and they use `current` I'm guessing we can't make those changes until the version bump happen?

Good point, I've updated the URLs to master and switched the target.
In general, we try to get around this fragility by using the @doc_id and @ext_doc_id keywords when we're linking within the specifications, which are resolved from https://github.com/elastic/elasticsearch-specification/blob/main/specification/_doc_ids/table.csv 
The overlay files can't make use of that table, however, so until we get this info pushed into the spec it'll need manual maintenance... but I'll use version-specific URLs instead of current to help mitigate the maintenance.

leemthompo commented on docs/overlays/elasticsearch-shared-overlays.yaml:
```diff
@@ -35,12 +35,13 @@ actions:
           externalDocs:
-            url: https://www.elastic.co/guide/en/enterprise-search/current/connectors-tutorial-api.html
+            url: https://www.elastic.co/guide/en/elasticsearch/reference/master/es-connectors-tutorial-api.html
             description: To get started with Connector APIs, check out the tutorial.
```
```suggestion
            description: Connector API tutorial
```

leemthompo commented on docs/overlays/elasticsearch-shared-overlays.yaml:
```diff
@@ -35,12 +35,13 @@ actions:
-            * Connector clients are self-managed on your infrastructure
+            * Elastic managed connectors (Native connectors) are a managed service on Elastic Cloud
+            
+            * Self-managed connectors (Connector clients) are self-managed on your infrastructure
```
```suggestion
            * Self-managed connectors (Connector clients) are self-managed on your infrastructure.
```

leemthompo reviewed (commented):
Found couple little things in preview! 🔍
